arkcompiler/ets_frontend/es2panda/util/helpers.cpp

```diff
@@ -630,10 +630,11 @@ SignedNumberLiteral Helpers::GetSignedNumberLiteral(const ir::Expression *expr)
 
 void Helpers::OptimizeProgram(panda::pandasm::Program *prog,  const std::string &inputFile)
 {
+    // The possibility of handling variables that may not be used
     std::map<std::string, size_t> stat;
-    std::map<std::string, size_t> *statp = &stat;
+    [[maybe_unused]] std::map<std::string, size_t> *statp = &stat;
```

Comment:
Do we need to explicitly enable `c++17`？

Reply:
or a later stdc++ version

Reply:
![image](https://github.com/ark-compiler/monorepo/assets/96290967/10832df5-9f89-41a9-9ffa-73f7231cde1d)

In compilation, GN is configured to explicitly use C++17, maybe we need do nothing. 

`[[maybe_unused]]` is used because there is a `[-Werror,-Wunused-variable]` configuration, which will report an error if the following macro is not selected. **In fact, I don't know if adding `[[maybe_unused]]` will cause subsequent problems, just to make the compilation work**.

Reply:
> Do we need to explicitly enable `c++17`？

![image](https://github.com/ark-compiler/monorepo/assets/96290967/10832df5-9f89-41a9-9ffa-73f7231cde1d)

In compilation, GN is configured to explicitly use C++17, maybe we need do nothing. 

`[[maybe_unused]]` is used because there is a `[-Werror,-Wunused-variable]` configuration, which will report an error if the following macro is not selected. **In fact, I don't know if adding `[[maybe_unused]]` will cause subsequent problems, just to make the compilation work**.